src/ChunkMap.cpp

```diff
@@ -1814,10 +1814,21 @@ void cChunkMap::DoExplosionAt(double a_ExplosionSize, double a_BlockX, double a_
 				a_Entity.TakeDamage(dtExplosion, nullptr, static_cast<int>((1 / DistanceFromExplosion.Length()) * 6 * ExplosionSizeInt), 0);
 			}
 
-			// Apply force to entities around the explosion - code modified from World.cpp DoExplosionAt()
-			DistanceFromExplosion.Normalize();
-			DistanceFromExplosion *= ExplosionSizeInt * ExplosionSizeInt;
-			a_Entity.AddSpeed(DistanceFromExplosion);
+			double Length = DistanceFromExplosion.Length();
+			if (Length <= ExplosionSizeInt)  // Entity is impacted by explosion
+			{
+				float EntityExposure = a_Entity.GetExplosionExposureRate(ExplosionPos, ExplosionSizeInt);
+
+				// Exposure reduced by armor
+				EntityExposure = static_cast<float>(EntityExposure * (1.0 - a_Entity.GetEnchantmentBlastKnockbackReduce()));
+
+				double Impact = (1 - (Length / ExplosionSizeInt / 2)) * EntityExposure;
```

Comment:
actually, about this impact calculation, what is describled in wiki is kind confuse.
https://minecraft.gamepedia.com/Explosion

```
For every entity within a 2×[power] block sphere of the explosion center, the impact is (1-[distance from explosion/power/2])×[exposure] (see section below on exposure)
```

I am not sure about how impact should calculate, 
* 1 - (distance / (power / 2))
* 1 - ((distance / power) / 2)

I prefer first one, because it is more make sense and impact strength will be more smooth at the edge of explosion effect box.

Am I correct?